pkg/manager/internal.go

```diff
@@ -619,9 +619,9 @@ func (cm *controllerManager) Engage(ctx context.Context, cl cluster.Cluster) err
 	// be reentrant via noop
 	cm.engagedClustersLock.RLock()
 	if _, ok := cm.engagedClusters[cl.Name()]; ok {
-		cm.engagedClustersLock.RUnlock()
 		return nil
 	}
+	cm.engagedClustersLock.RUnlock()
```

Comment:
great find! but don't we need both? RUnlock on return, and RUnlock for the else case?

Reply:
Yes! I missed that. I have also update the `fleet` example with the same error [here](https://github.com/ivanape/controller-runtime/blob/sttts-cluster-support/examples/fleet/main.go#L172-L177).